### PR TITLE
support runner.arch

### PR DESCRIPTION
**Pull request by datosh**
Signed-off-by: Fabian Kammel <fk@edgeless.systems>

### CODE REVIEW

**Comment by datosh** [comment]
I have looked at https://github.com/nektos/act/issues/171 on how to add support. After running `make build` and using the produced binary I am still not able to read from `${{ runner.arch }}` am I missing something?

**Comment by datosh** [comment]
> Yes it is defined somewhere else, here
> 
> https://github.com/nektos/act/blob/3364f95569d8a39d440b18af69d29421fcc77b55/pkg/runner/expression.go#L93-L97
> 
> 
> and here
> https://github.com/nektos/act/blob/3364f95569d8a39d440b18af69d29421fcc77b55/pkg/runner/expression.go#L48-L52

Thanks, I was able to change them to static values for `X64` and got my pipeline working locally 🥳 

> I don't think hardcoding X64 as RUNNER_ARCH is a good idea, arm64 is also valid. I suggest to derive the value from `container.GetHostInfo(ctx) ` like done here, however on m1 mac books act supports X64 via qemu with special flags....
> 
> https://github.com/nektos/act/blob/3364f95569d8a39d440b18af69d29421fcc77b55/cmd/root.go#L187

I changed the code to fetch values from Docker as you suggested, but I have few open points:

* Architecture mapper is incomplete. Can someone provide supported values for the architecture returned by docker so we can translate them properly? This is necessary, since the values provided by Docker and expected by GitHub Actions are different.
* Testability of my solution is poor, I would prefer passing an interface which can be subbed during testing. Would this fit to your coding conventions? Any pointers on how to proceed?

**Comment by datosh** [comment]
gentle pint @ChristopherHX

File: pkg/runner/expression.go
Comment (datosh) [reply pkg/runner/expression.go:54]: > I'm not shure if we need to use docker inspect on the jobcontainer to get reliable the arch of the container, but this can wait until the HostExecution Environment is merged and we have full access to the container structure.

Okay cool - thanks for getting back. 
I am fine with using my fork for now 🙂

**Comment by datosh** [comment]
> I'm not shure if we need to use docker inspect on the jobcontainer to get reliable the arch of the container, but this can wait until the HostExecution Environment is merged and we have full access to the container structure.

Okay cool - thanks for getting back. 
I am fine with using my fork for now 🙂

**Review by KnisterPeter — APPROVED** [review]

**Comment by datosh** [comment]
> I fixed the not goimported linting error of my proposed changes, I wish the CI does this automatically during merge. 
> 
> LGTM, m1 mac's using the `container-arch` option with qemu will get arm64 and not x64.
> If this is a problem then we should wait for the host environment PR, to move the runner context into the extended container interface.

Awesome. Thanks for your help!

**Comment by datosh** [comment]
Just to make sure I am not overlooking something. I am not authorized to merge, so I guess you will do this in due time @KnisterPeter @ChristopherHX  ?

**Review by cplee — APPROVED** [review]

**Comment by KnisterPeter** [comment]
A third approval is required

**Pull request merged**
